ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/CrosslinkCommitteeUtil.java

```diff
@@ -14,31 +14,138 @@
+import static tech.pegasys.artemis.datastructures.util.BeaconStateUtil.min;
+import static tech.pegasys.artemis.datastructures.util.ValidatorsUtil.get_active_validator_indices;
 
 public class CrosslinkCommitteeUtil {
-  // Return the number of committees in the previous epoch of the given ``state`
-  public static int get_previous_epoch_committee_count(BeaconState state) {
-    List<Validator> previous_active_validators =
-        ValidatorsUtil.get_active_validators(
-            state.getValidator_registry(), state.getPrevious_shuffling_epoch());
-    return previous_active_validators.size();
+
+  /**
+   * Return the shuffled validator index corresponding to ``seed`` (and ``index_count``).
+   *
+   * @param index
+   * @param index_count
+   * @param seed
+   * @return
+   *
+   * @see <a>https://github.com/ethereum/eth2.0-specs/blob/v0.7.1/specs/core/0_beacon-chain.md#get_shuffled_index</a>
+   */
+  public static Integer get_shuffled_index(int index, int index_count, Bytes32 seed){
+    checkArgument(index < index_count, "CrosslinkCommitteeUtil.get_shuffled_index1");
+    checkArgument(index_count <= Math.pow(2, 40), "CrosslinkCommitteeUtil.get_shuffled_index2");
+
+    //Swap or not (https://link.springer.com/content/pdf/10.1007%2F978-3-642-32009-5_1.pdf)
+    //See the 'generalized domain' algorithm on page 3
+    for(int round = 0; round < SHUFFLE_ROUND_COUNT; round++){
+      Bytes roundBytes = int_to_bytes(round, 1);
+
+      Bytes32 pivotHashValue = Hash.sha2_256(Bytes.concatenate(seed, roundBytes));
+      Bytes pivotHashSubArray = Bytes.wrap(ArrayUtils.subarray(pivotHashValue.toArray(), 0, 8));
+      int pivot = ((int)bytes_to_int(pivotHashSubArray)) % index_count;
+
+      int flip = (pivot + index_count - index) % index_count;
+      long position = max(UnsignedLong.valueOf(index), UnsignedLong.valueOf(flip)).longValue();
+
+      Bytes sourceBytes = Bytes.concatenate(seed, roundBytes, int_to_bytes(Math.floorDiv(position, 256l), 4));
+      Bytes32 source = Hash.sha2_256(sourceBytes);
+
+      int sourceByteIndex = toIntExact(Math.floorDiv((position % 256), 8l));
+      byte byteValue = source.get(sourceByteIndex);
+      int bit = (byteValue >> (position % 8)) % 2;
+      index = (bit == 1) ? flip : index;
+    }
+    return index;
+  }
+
+  /**
+   * Computes indices of a new committee based upon the seed parameter
+   *
+   * @param indices
+   * @param seed
+   * @param index
+   * @param count
+   * @return
+   *
+   * @see <a>https://github.com/ethereum/eth2.0-specs/blob/v0.7.1/specs/core/0_beacon-chain.md#compute_committee</a>
+   */
+  public static List<Integer> compute_committee(List<Integer> indices, Bytes32 seed, int index, int count){
+    int start = Math.floorDiv(indices.size() * index, count);
+    int end = Math.floorDiv(indices.size() * (index + 1), count);
+    for(int i = start; i < end; i++)indices.set(i, get_shuffled_index(i, indices.size(), seed));
```

Comment:
We should replace this line with: return IntStream.range(start, end).map(i -> index.get(get_shuffled_index(i, indices.size(), seed))).collect(Collectors.toList());